app/services/markdown_processor/fixer/base.rb

```diff
@@ -0,0 +1,95 @@
+module MarkdownProcessor
+  module Fixer
+    # Services here in the Fixer module should inherit from this base class.
+    # #call is implemented here so other services only need to define a
+    # METHODS constant that is an Array of symbols referencing other methods
+    # found here.
+    #
+    # For example
+    # METHODS = %i[add_quotes_to_tile add_quotes_to_description]
+    class Base
+      FRONT_MATTER_DETECTOR = /-{3}.*?-{3}/m.freeze
+
+      def self.call(markdown)
+        return unless markdown
+
+        self::METHODS.reduce(markdown) { |acc, elem| public_send(elem, acc) }
```

Comment:
This "approach" is similar to how to we handle the interface with Elasticsearch under `/search` where each class defines a set of constants and the `base` class handles the logic.

Reply:
I had the same concern about the search classes, IMHO the list of methods should come from a method, even if that just wraps the constant. Methods are much easier to override etc., also while testing.

```suggestion
        fix_methods.reduce(markdown) { |acc, elem| public_send(elem, acc) }
```

```ruby
def fix_methods
  METHODS
end
```

Reply:
Good idea, updated!